app/models/product_importer.rb

```diff
@@ -177,7 +181,12 @@ def count_existing_products
           AND spree_variants.deleted_at IS NULL', supplier_id).
           count
 
-        @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        if @supplier_products[:by_supplier][supplier_id]
+          @supplier_products[:by_supplier][supplier_id][:existing_products] = products_count
+        else
+          @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        end
```

Comment:
I thought so initially, but there are two keys being set in the same space in the hash at different times, so I found it didn't work without this conditional. I'll have another look and see if it can be refactored a bit now, that it's finished...